client/src/modules/stock/entry/entry.js

```diff
@@ -3,45 +3,49 @@ angular.module('bhima.controllers')
 
 // dependencies injections
 StockEntryController.$inject = [
-  'DepotService', 'InventoryService', 'NotifyService',
-  'SessionService', 'util', 'bhConstants', 'ReceiptModal', 'PurchaseOrderService',
-  'StockFormService', 'StockService', 'StockModalService', 'uiGridGroupingConstants',
-  'uiGridConstants', 'Store',
+  'DepotService', 'InventoryService', 'NotifyService', 'SessionService', 'util',
+  'bhConstants', 'ReceiptModal', 'PurchaseOrderService', 'StockFormService',
+  'StockService', 'StockModalService', 'uiGridConstants', 'Store', 'appcache',
 ];
 
 /**
- * StockEntryController
- * @description This controller is responsible to handle stock entry module
+ * @class StockEntryController
+ *
+ * @description
+ * This controller is responsible to handle stock entry module.
  */
-function StockEntryController(Depots, Inventory, Notify,
-  Session, util, bhConstants, ReceiptModal, Purchase,
-  StockForm, Stock, StockModal, uiGridGroupingConstants, uiGridConstants, Store) {
+function StockEntryController(
+  Depots, Inventory, Notify, Session, util, bhConstants, ReceiptModal, Purchase,
+  StockForm, Stock, StockModal, uiGridConstants, Store, AppCache
+) {
   var vm = this;
   var inventoryStore;
 
+  var cache = new AppCache('StockEntry');
+
   var mapEntry = {
     purchase    : { find : findPurchase, submit : submitPurchase },
-    donation    : { find : angular.noop, submit : angular.noop }, // donation is not yet implemented
+    donation    : { find : angular.noop, submit : angular.noop },  // donation is not yet implemented
     integration : { find : handleIntegrationSelection, submit : submitIntegration },
     transfer_reception : { find : findTransfer, submit : submitTransferReception },
   };
 
   vm.Stock = new StockForm('StockEntry');
-  vm.depot = {};
   vm.movement = {};
 
   // bind methods
   vm.itemIncrement = 1;
   vm.enterprise = Session.enterprise;
   vm.maxLength = util.maxLength;
+  vm.maxDate = new Date();
```

Comment:
It is seems like ```vm.maxDate``` is never used

Reply:
You are right!  It isn't ever used, but it should be - see https://github.com/IMA-WorldHealth/bhima-2.X/pull/2097 which fixes this in at least one module.